cms-adapter/src/Rsbc.Dmf.CaseManagement.Service/Services/DocumentService.cs

```diff
@@ -119,6 +119,88 @@ public async override Task<GetDmerReply> GetDmer(CaseIdRequest request, ServerCa
+                result.ResultStatus = ResultStatus.Success;
+                
+            }
+            catch (Exception ex)
+            {
+                result.ResultStatus = ResultStatus.Fail;
+                result.ErrorDetail = ex.Message;
+            }
+
+            return result;
+        }
+
+        // TODO driver id is no longer relevant and method name should be changed
+        public async override Task<GetDriverAndCaseDocumentsReply> GetDriverAndCaseDocuments(GetDriverAndCaseDocumentsRequest request, ServerCallContext context)
+        {
+            var result = new GetDriverAndCaseDocumentsReply();
+
+            try
+            {
+                 var loginId = request.LoginId;
+                var caseId = request.CaseId;
+                
+                var documents = _documentManager.GetDriverAndCaseDocuments(caseId, loginId);
+                var mappedDocuments = _mapper.Map<IEnumerable<Document>>(documents);
+                result.Items.AddRange(mappedDocuments);
+                result.ResultStatus = ResultStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                result.ResultStatus = ResultStatus.Fail;
+                result.ErrorDetail = ex.Message;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Update ClaimDmer
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async override Task<UpdateClaimReply> UpdateClaimDmer(UpdateClaimRequest request, ServerCallContext context)
+        {
+            var result = new UpdateClaimReply();
+
+            try
+            {
+                var loginId = Guid.Parse(request.LoginId);
+                var documentId = Guid.Parse(request.DocumentId);
+                var documents = _documentManager.UpdateClaimDmer(loginId, documentId);
+                var mappedDocuments = _mapper.Map<Document>(documents);
+                //result.Items.AddRange(mappedDocuments);
+                result.ResultStatus = ResultStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                result.ResultStatus = ResultStatus.Fail;
+                result.ErrorDetail = ex.Message;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Update UnClaimDmer
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public async override Task<UpdateClaimReply> UpdateUnClaimDmer(UpdateClaimRequest request, ServerCallContext context)
+        {
+            var result = new UpdateClaimReply();
+
+            try
+            {
+                var loginId = Guid.Parse(request.LoginId);
+                var documentId = Guid.Parse(request.DocumentId);
+                var documents = _documentManager.UpdateUnClaimDmer(loginId, documentId);
```

Comment:
should be a singular document not a list

Reply:
Fixed this for unclaimed Method